library/src/main/java/com/datatorrent/lib/iteration/nDelayOperator.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.lib.iteration;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.validation.constraints.Min;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.Operator;
+import com.datatorrent.lib.util.WindowDataManager;
+import com.datatorrent.netlet.util.DTThrowable;
+
+/**
+ * This operator is used in iteration and can do n window delay.
+ * Required delay needs to be set in the constructor, default delay is 1.
+  *
+ * @displayName nDelayOperator
+ * @category iteration
+
+ */
+public class nDelayOperator<T> implements Operator.DelayOperator, Operator.CheckpointListener
```

Comment:
Sorry for nitpicking but the name is not according to Java Convention of naming classes